src/snowflake/connector/result_batch.py

```diff
@@ -651,21 +664,45 @@ def _get_arrow_iter(
         """Returns an iterator for this batch which yields a pyarrow Table"""
         return self._create_iter(iter_unit=IterUnit.TABLE_UNIT, connection=connection)
 
-    def to_arrow(
-        self, connection: Optional["SnowflakeConnection"] = None
-    ) -> Optional[Table]:
+    def _create_emtpy_table(self) -> Table:
+        """Returns emtpy Arrow table based on schema"""
+        if installed_pandas:
```

Comment:
I don't think this is necessary inside of this function
```suggestion
```

Reply:
This is the safeguard to protect against the unresolved imports from pyarrow when lazy evaluating.